src/main/kotlin/com/cognifide/gradle/aem/common/instance/InstanceManager.kt

```diff
@@ -185,8 +195,62 @@ open class InstanceManager(val aem: AemExtension) {
         }
     }
 
-    fun examineRunningOther(instances: Collection<Instance>) {
-        val running = instances.filterIsInstance<LocalInstance>().filter { it.runningOther }
+    /**
+     * Checks if local instances defined are meeting prerequisites before performing any other operations.
+     *
+     * Assumes that instances could not be running yet.
+     */
+    fun examinePrerequisites(instances: Collection<Instance> = aem.instances) {
+        val localInstances = instances.filterIsInstance<LocalInstance>()
+        examineJavaCompatibility(localInstances)
+        examineRunningOther(localInstances)
+    }
+
+    /**
+     * Defines compatibility related to AEM versions and Java versions.
+     *
+     * AEM version is definable as range inclusive at a start, exclusive at an end.
+     * Java Version is definable as list of supported versions comma delimited.
+     */
+    val javaCompatibility = aem.obj.map<String, String> {
+        convention(mapOf(
+                "6.0.0-6.5.0" to "1.7,1.8",
+                "6.5.0-6.6.0" to "1.8,11"
+        ))
+        aem.prop.map("instance.javaCompatibility")?.let { set(it) }
+    }
+
+    @Suppress("NestedBlockDepth")
+    fun examineJavaCompatibility(instances: Collection<LocalInstance> = aem.localInstances) {
+        if (javaCompatibility.get().isEmpty()) {
+            return
+        }
+
+        val versionCurrent = JavaVersion.current()
+        val errors = mutableListOf<String>()
```

Comment:
Avoid mutability.

Reply:
will try, however not easy. I need sth like mapNotNull here but there is a problem with type inference there. I could change it to fold but it will be effectively same.